Gui list

markhelmus opened the pull request.

markhelmus committed:
Basic GUI:
- App has a list of peripherals
- ListItem consists of clickable buttons
- Peripheral 'Lock' can communicate with a server

markhelmus committed:
Added build.gradle

markhelmus committed:
Added all gradle files

markhelmus committed:
Added full project

FrankTN commented:
You should probably remove the .idea files and update the .gitignore to prevent the project from getting messy

FrankTN commented on Hestia/app/src/main/java/com/rugged/application/hestia/Client.java:
```diff
@@ -0,0 +1,66 @@
+        // Strings line "abc...\n" -> the "\n" would be visible
+        jsonObj.put("id", id);
+        jsonObj.put("action", action);
+        this.writeJSONToServer(jsonObj);
```
This should comply with the JSON object as defined on basecamp, so:
`
{
  "action": {
    "id": 1,
    "type": "Lock"
  }
}
`

FrankTN reviewed (changes requested):
In general, this looks good. There are a lot of hardcoded things which need to change at a later stage.

My main concern is with the JSON specification, which does not agree completely with the one published on Basecamp.

I also recommend adding comments in critical sections, to increase readability.

suzannemaquelin commented on Hestia/app/src/main/java/com/rugged/application/hestia/PeripheralFragment.java:
```diff
@@ -0,0 +1,105 @@
+        offButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                new SendJSONFile2().execute("82.73.173.179", "openLock");
```
For the off button, shouldn't it be "closeLock"?

suzannemaquelin commented on Hestia/app/src/main/java/com/rugged/application/hestia/PeripheralLab.java:
```diff
@@ -0,0 +1,52 @@
+import java.util.List;
+import java.util.UUID;
+
+public class PeripheralLab {
```
It's probably just me, but I don't understand what peripheralLab and sPeripheralLab are

FrankTN replied:
I had the same thing, its probably a good idea to add some comments in these parts.

markhelmus replied:
I added some explaining comments. Hopefully it is understandable what those classes are doing. If not, please let me know.

suzannemaquelin reviewed (changes requested):
I think you made a mistake with "openLock". For the off button it should be closeLock

Feikoritsema committed:
Changed to unLock and changed JSON action to type.

Feikoritsema reviewed (approved):
It seems weird to review my own code, I read through it once more and approve.

markhelmus committed:
Changed JSONObject.
Note: sending {"a": {"b": 1, "c": 2}} is defined as
JSONObject jsonObj = new JSONObject();
jsonObj.put("c", 2);
jsonObj.put("b", 1);
JSONObject jsonA = new JSONObject();
jsonA.put("a", jsonObj);

markhelmus committed:
Added explaining comments to multiple classes

markhelmus committed:
Updated gitignore
